api/src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecution.java

```diff
@@ -158,8 +158,14 @@ public String getUrl() throws IOException {
      */
     public abstract void interrupt(Result r, CauseOfInterruption... causes) throws IOException, InterruptedException;
 
+    /**
+     * Add a listener to changes in the flow graph structure.
+     * @param listener a listener to add
+     */
     public abstract void addListener(GraphListener listener);
 
+    public /*abstract*/ void removeListener(GraphListener listener) {}
```

Comment:
Needs Javadoc and "TODO: make abstract" for the next backward-incompatible version